### PR TITLE
server: migrate test-infra to testify for http_handler_test.go

### LINKED ISSUE #27804: migrate test-infra to testify for `server/http_handler_test.go`
This is a subtask of #26863

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #27804 .

Signed-off-by: tison <wander4096@gmail.com>

### Release note

```release-note
None
```

/cc @morgo @disksing

### CODE REVIEW

**Review by disksing — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

File: server/http_handler_test.go [hunk 1]
```diff
@@ -320,282 +313,268 @@ func (ts *HTTPHandlerTestSuite) regionContainsTable(c *C, regionID uint64, table
...
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	ts.tidbdrv = NewTiDBDriver(ts.store)
 
 	cfg := newTestConfig()
 	cfg.Store = "tikv"
 	cfg.Port = 0
 	cfg.Status.StatusPort = 0
 	cfg.Status.ReportStatus = true
+	cfg.Socket = fmt.Sprintf("/tmp/%s.sock", t.Name())
```
Comment (bb7133) [inline server/http_handler_test.go:462]: Is there any reason cfg.Socket is added in the server for tests? @tisonkun 

I guess it caused https://github.com/pingcap/tidb/issues/30642

Comment (tisonkun) [reply server/http_handler_test.go:462]: IIUC it's for using different socket file when running parallel tests.

**Pull request merged**